pkg/operator/aws_platform_service_location/controller.go

```diff
@@ -38,12 +49,94 @@ func NewController(operatorClient operatorv1helpers.OperatorClient, infraClient
 }
 
 func (c AWSPlatformServiceLocationController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	currentInfra, err := c.infraClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	obj, err := c.infraLister.Get("cluster")
 	if errors.IsNotFound(err) {
-		// log not found errors, but don't degrade?
-		klog.Warningf("%w", err)
+		syncCtx.Recorder().Warningf("AWSPlatformServiceLocationController", "Required infrastructures.%s/cluster not found", configv1.GroupName)
 		return nil
 	}
-	klog.Info(spew.Sprintf("current state: %s", currentInfra.Status))
+	if err != nil {
+		return err
+	}
+
+	currentInfra := obj.DeepCopy()
+	var platformName configv1.PlatformType
+	if pstatus := currentInfra.Status.PlatformStatus; pstatus != nil {
+		platformName = pstatus.Type
+	}
+	if platformName == "" {
+		syncCtx.Recorder().Warningf("AWSPlatformServiceLocationController", "Falling back to deprecated status.platform because infrastructures.%s/cluster status.platformStatus.type is empty", configv1.GroupName)
+		platformName = currentInfra.Status.Platform
+	}
+
+	if platformName != configv1.AWSPlatformType {
+		return nil // nothing to do here.
+	}
+
+	if !(currentInfra.Spec.PlatformSpec != nil &&
+		currentInfra.Spec.PlatformSpec.Type == configv1.AWSPlatformType &&
+		currentInfra.Spec.PlatformSpec.AWS != nil &&
+		len(currentInfra.Spec.PlatformSpec.AWS.ServiceEndpoints) > 0) {
+		return nil // nothing to do here
+	}
+
+	services := currentInfra.Spec.PlatformSpec.AWS.ServiceEndpoints
```

Comment:
deep copy please

Reply:
https://github.com/openshift/cluster-config-operator/pull/119/commits/d957d968accfb543daecb6c1d7e10ecfd8584467#diff-51a3ce64d08707b029401c01f6a81ac0R61

already deepcopies.

Reply:
> [d957d96#diff-51a3ce64d08707b029401c01f6a81ac0R61](https://github.com/openshift/cluster-config-operator/commit/d957d968accfb543daecb6c1d7e10ecfd8584467#diff-51a3ce64d08707b029401c01f6a81ac0R61)
> 
> already deepcopies.

Given the assignment back later, I'd prefer not to shallow copy this.  I think it's just .DeepCopy, right?

Reply:
`ServiceEndpoints` is a slice, so no DeepCopy.

i could DeepCopy the entire currentInfra again..
`currentInfra.DeepCopy().Spec.PlatformSpec.AWS.ServiceEndpoints` ?? @deads2k 

Can you provide a little background on why you think deepcopy here would be necessary?

Reply:
> `ServiceEndpoints` is a slice, so no DeepCopy.
> 
> i could DeepCopy the entire currentInfra again..
> `currentInfra.DeepCopy().Spec.PlatformSpec.AWS.ServiceEndpoints` ?? @deads2k
> 
> Can you provide a little background on why you think deepcopy here would be necessary?

mutating the input means that if you later use it or diff it somewhere, it won't be what you expect it to be.  Makes it hard for me to reason about it's state during this review.